lib/access-context.js

```diff
@@ -138,17 +146,17 @@ AccessContext.prototype.getUserId = function() {
 
     // the principalType must either be 'USER'
     if (p.type === Principal.USER) {
-      return p.id;
+      return {id: p.id, principalType: p.type};
     }
 
     // or permit to resolve a valid user model
     var userModel = this.registry.findModel(p.type);
     if (!userModel) continue;
     if (userModel.prototype instanceof BaseUser) {
-      return p.id;
+      return {id: p.id, principalType: p.type};
     }
   }
-  return null;
+  return {};
```

Comment:
if we do this, then this will break the `getUserId()` method for which is seems normal to return null according to the tests. Should we then modify the getUserId method to return null when getUser() returns null? 
If so, then there's not much need to not return `{}` in the first place